container_linux.go

```diff
@@ -249,6 +257,124 @@ func (c *linuxContainer) NotifyOOM() (<-chan struct{}, error) {
 	return notifyOnOOM(c.cgroupManager.GetPaths())
 }
 
+// XXX debug support, remove when debugging done.
+func addArgsFromEnv(evar string, args *[]string) {
+	if e := os.Getenv(evar); e != "" {
+		for _, f := range strings.Fields(e) {
+			*args = append(*args, f)
+		}
+	}
+	fmt.Printf(">>> criu %v\n", *args)
+}
+
+func (c *linuxContainer) Checkpoint() error {
+	c.m.Lock()
+	defer c.m.Unlock()
+	dir := filepath.Join(c.root, "checkpoint")
+	// Since a container can be C/R'ed multiple times,
+	// the checkpoint directory may already exist.
+	if err := os.Mkdir(dir, 0655); err != nil && !os.IsExist(err) {
+		return err
+	}
+	args := []string{
+		"dump", "-v4",
+		"-D", dir,
+		"-o", "dump.log",
+		"--root", c.config.Rootfs,
+		"--manage-cgroups", "--evasive-devices",
+		"-t", strconv.Itoa(c.initProcess.pid()),
+	}
+	for _, m := range c.config.Mounts {
+		if m.Device == "bind" {
+			args = append(args,
+				"--ext-mount-map", fmt.Sprintf("%s:%s", m.Destination, m.Destination))
+		}
+	}
+	addArgsFromEnv("CRIU_C", &args) // XXX debug
+	if err := exec.Command(c.criuPath, args...).Run(); err != nil {
```

Comment:
@LK4D4 @vishh 
CRIU logs everything in the log files and generally doesn't output.